src/transformers/trainer.py

```diff
@@ -2238,10 +2246,13 @@ def evaluation_loop(
 
         model = self._wrap_model(self.model, training=False)
 
-        # if full fp16 is wanted on eval and this ``evaluation`` or ``predict`` isn't called while
-        # ``train`` is running, halve it first and then put on device
-        if not self.is_in_train and self.args.fp16_full_eval:
-            model = model.half().to(self.args.device)
+        # if full fp16 or bf16 eval is wanted and this ``evaluation`` or ``predict`` isn't called
+        # while ``train`` is running, cast it to the right dtype first and then put on device
+        if not self.is_in_train:
+            if args.fp16_full_eval:
```

Comment:
Couldn't lines 2251 to 2255 be replaced with?
```python
if not self.is_in_train and (args.fp16_full_eval or args.bf16_full_eval):
    model = model.to(dtype=self.amp_dtype, device=args.device)
```

Reply:
`self.amp_dtype` is None here, this the case where there is no AMP.

Reply:
Ohh… my bad.

Reply:
Please worry not, it's very hard to keep track of all the dozens of variables, 

but that was a great suggestion otherwise!